app/controllers/concerns/foreman_kernel_care/foreman_tasks.rb

```diff
@@ -8,11 +8,11 @@ def callback
         # We need to call .to_unsafe_h to unwrap the hash from ActionController::Parameters
         callback = payload[:callback]
         foreman_task = foreman_tasks.find { |task| task.id == callback[:task_id] }
-        if foreman_task.action.include?('Get patched kernel version')
-          version, release = payload[:data].to_unsafe_h['result'].first['output'].strip.split('-')
-          job_invocation = ::JobInvocation.where(:task_id => foreman_task.parent_task_id).first
-          job_invocation.targeting.hosts.each { |host| host.update_kernel_version(version, release) }
-        end
+        next unless foreman_task.action.include?('Get patched kernel version')
+        kcare, kernel = payload[:data].to_unsafe_h['result'].first['output'].split("\n")
+        version, release = kcare.strip.split('-')
```

Comment:
if no "-" is found, release would be nil. later on, update_kernel_version with"release" is called and a version like "1.2.3-" would be created. Some more checks would make sense (and protection with a test)